### ISSUE TITLE
Create User Manual with screenshots and videos for Plone 6

### ISSUE THREAD

**Issue by stevepiercy** [plone/Products.CMFPlone#3987]
## Abstract

The User Manual for Plone 6 does not exist, except for a small part of Volto in https://6.docs.plone.org/volto/user-manual/index.html.

We refer users to Plone 5 documentation [Working with Content](https://5.docs.plone.org/working-with-content/index.html) for Classic UI, but it is out of date.

Writing narrative text for a user manual is time consuming and rarely gets translated. If it gets translated, it is immediately out of date.

However through the use of Cypress as an automated test runner, we can take screenshots and videos of the frontends in any language, and incorporate these into a translated User Manual.

## Motivation

User Manual documentation has been neglected in Plone 6 since its release. This shows how badly we disregard the end user.

## Assumptions

Writing narrative documentation of a User Manual is intimidating and tedious. A picture or video is worth a thousand words, especially for those who don't write English as their first language.

The frontend rapidly changes, and we can update documentation automatically on every pull request.

Both Cypress and Playwright perform similar functions, but we already have all acceptance tests for Volto written with Cypress and a large base of developers with experience with Cypress. Supposedly Classic UI has opted to use Playwright, and there was some effort invested into this, but the author of this PLIP is not aware of the status of that effort.

A pull request was opened by @tkimnguyen https://github.com/plone/documentation/pull/1547 during PloneConf 2023, but it used old screenshots, its content was not reviewed for accuracy, it is in a Draft state, and has stalled.

## Proposal and Implementation

Investigate the use of automated tools to record videos and screenshots, and save them into the appropriate `_static` directory and repository.

- [Cypress Studio](https://docs.cypress.io/app/guides/cypress-studio)
- [Playwright test generator](https://playwright.dev/docs/codegen)
- [`shot-scraper`](https://shot-scraper.datasette.io/en/stable/index.html)
- [Dubble](https://dubble.so/). Has an [API](https://dubble.readme.io/reference/getcollections) for [paid plans starting at $18/month for 3 creators](https://dubble.so/pricing).
- [Argos](https://argos-ci.com/) for visual regression testing in CI, also takes screenshots
- [heroshot](https://heroshot.sh/) which now has integration with Sphinx per https://github.com/omachala/heroshot/issues/34.

Cypress and Playwright are designed for running acceptance tests and not enhancing documentation, but they still might be useful.

### Goals

- Full automation that is repeatable.
- Integration with GitHub workflows.
- Generate narrative text from interacting with web page elements, such as "Click the `Save` button".
- Save the recorded steps as code into the appropriate repository for collaborative editing and execution.

## Deliverables

- [ ] Automation of updated screenshots and videos on every merged pull request through GitHub Workflows.
- [ ] Port of Plone 5 user manual to Plone 6, converted to MyST with SEO headers and reviewed for accuracy.

## Risks

We need to avoid creating too many videos and screenshots. A presentation and blog post provide excellent guidance.

- https://docs.google.com/presentation/d/1JC05eK1jx9QdbqUkQCQL_vpsauvv-mhORwaV2FXPQqQ/edit#slide=id.g3416219aa79_0_297
- https://thisisimportant.net/posts/screenshots-in-documentation/

## Participants

Steve Piercy, @stevepiercy
Alin Voinea @avoinea

**Comment by tkimnguyen** [comment]
The Content Editing training guide covers what we show people who are new to Volto. It would be lovely to repurpose that content. 

* https://training.plone.org/content-editing/index.html 
* https://github.com/plone/training/tree/main/docs/content-editing

**Comment by tisto** [comment]
@stevepiercy This might be one of the most important PLIPs that I've seen in the past years. So +1 from me.

I am going through all PLIPs and trying to assign them to milestones. Is Plone 6.2 (end of 2025, early 2026) a realistic milestone? What steps are needed? How can we help?

**Comment by stevepiercy** [comment]
@tisto I'd like to discuss at PloneConf 2025 how to move forward. My opinion has evolved since I proposed this PLIP.

Although the idea of automating screenshots and videos is admirable, it's technically complicated. Ideally, acceptance tests in Volto could also automatically generate the visuals, but we found that it's easier said than done.

Manual updating of visuals is not sustainable.

Additionally, amongst the documentation community, there is a strong opinion that if your application requires screenshots and videos to explain how to perform a task, instead of just plain old text, then your app is too complicated. I respect these professionals' opinions, and have come around to agree with them. A visual for every click or keystroke is just too much, and actually makes documentation difficult to read. It's better to provide visuals when words become too much and "a picture is worth a thousand words".

I hope to be free of critical Treasurer duties after PloneConf 2025, and I'd like to apply for grants to fund the work. It's a huge task that just won't get done without paying someone to do it. Unfortunately, Google ended Season of Docs in 2025, and I have no idea where else to look for possible funding sources.

**Comment by tisto** [comment]
@stevepiercy I agree that paying someone to write those docs makes a lot of sense. I can both imagine chipping in and providing people who help with that task. However, someone would need to take the lead and organize such an effort. An open space at Ploneconf is a good idea!

**Comment by tkimnguyen** [comment]
It would be lovely to use/reuse as much as possible what is at https://training.plone.org/content-editing/index.html (and continues to be maintained).

**Comment by tkimnguyen** [comment]
It shows how to do a thing in quite some detail because there is so much in the UI that it is hard to describe using only words what a person should be doing.

**Comment by omachala** [comment]
Hi! I noticed heroshot was already mentioned in this PLIP - thanks for the consideration!

I'm the maintainer of [heroshot](https://heroshot.sh). It was built specifically for this use case - defining screenshots in a config file and regenerating them automatically when the UI changes.

The Sphinx integration mentioned in the PLIP is now stable. You can:
- Define all screenshots in a single config file
- Run one command to regenerate them all
- Integrate into CI/CD for automatic updates on PRs

I'd be happy to help set up a proof of concept for Plone 6 docs if that would be useful. The tool handles login flows, form interactions, and can wait for specific elements - which should cover most of the User Manual scenarios.

Happy to discuss at PloneConf or async!

**Comment by stevepiercy** [comment]
I've updated this PLIP to reflect that it's directed at Plone Site Editors and Managers, not just any user.
